commit/merkleroot/observation.go

```diff
@@ -440,6 +447,33 @@ func (o observerImpl) ObserveMerkleRoots(
 					return
 				}
 
+				if uint64(len(msgs)) != uint64(chainRange.SeqNumRange.End()-chainRange.SeqNumRange.Start()+1) {
```

Comment:
How about a `chainRange.Size()` function. It's in this repo now, I think there are a couple places in `SeqNumRange` that would use it also.